crates/vm/levm/src/precompiles.rs

```diff
@@ -1148,11 +1149,117 @@ fn point_evaluation(
+    consumed_gas: &mut u64,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    dbg!("entro", calldata.len(), consumed_gas.clone(), gas_for_call);
+    // Two inputs of 128 bytes are requiered
+    if calldata.len() != 256 {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    dbg!("pasa test length");
+
+    let mut first_point_x = calldata
+        .get(0..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut first_point_y = calldata
+        .get(64..128)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut second_point_x = calldata
+        .get(128..192)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    let mut second_point_y = calldata
+        .get(192..)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+
+    let zeros: [u8; 16] = [0_u8; 16];
+
+    // the first 16 bytes of any of the points MUST be all zeros
+    if first_point_x
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if first_point_y
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if second_point_x
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+    if second_point_y
+        .get(0..16)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?
+        != zeros
+    {
+        return Err(VMError::PrecompileError(PrecompileError::ParsingInputError));
+    }
+
+    first_point_x = first_point_x
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    first_point_y = first_point_y
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    second_point_x = second_point_x
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+    second_point_y = second_point_y
+        .get(16..64)
+        .ok_or(VMError::PrecompileError(PrecompileError::ParsingInputError))?;
+
+    dbg!(first_point_x, first_point_y, second_point_x, second_point_y);
+
+    let mut first_g1_points = Vec::new();
+    first_g1_points.extend_from_slice(first_point_x);
+    first_g1_points.extend_from_slice(first_point_y);
+
+    let first_g1_points: [u8; 96] = first_g1_points
+        .try_into()
+        .map_err(|_| VMError::Internal(InternalError::ConversionError))?;
+
+    let first_g1_group = G1Affine::from_uncompressed(&first_g1_points);
```

Comment:
This is not a group but an element of the group. Nevertheless, you can keep calling it `first_g1_point`.
```suggestion
    let first_g1_group = G1Affine::from_uncompressed(&first_g1_points);
```

Reply:
Done, [e0a0ebb](https://github.com/lambdaclass/ethrex/pull/1688/commits/e0a0ebb3bbee66d2691180f6cf8be034c45ab215)